cram_external_interfaces/cram_ik_interface/src/ik.lisp

```diff
@@ -134,3 +134,81 @@ If not valid solution was found, returns NIL."
+                    `(let ((sampling-values
+                             (append
+                              (list ,current-value)
+                              (loop
+                                for x = ,upper-limit
+                                  then  (- x ,resampling-step)
+                                until (<= x ,lower-limit)
+                                collect x)
+                              (list ,lower-limit)))
+                           ;; Calculating the pose offset for the corresponding joint value
+                           (pseudo-pose (lambda (offset)
+                                          (cram-tf:translate-pose
+                                           current-pose
+                                           (ecase ,resampling-axis
+                                             (:x :x-offset)
+                                             (:y :y-offset)
+                                             (:z :z-offset))
+                                           (- offset)))))
+                       (loop
+                         for value in sampling-values
+                         do
+                            (multiple-value-bind (solution-msg joint-values)
+                                ;; Checking if the arguments contain &body clause or not. 6 is
+                                ;; currently the number of arguments without including &body.
+                                ;; Update according to API changes
+                                (if (> ,form-length 6)
+                                    ;; If body is provided, call it with the current offseted
+                                    ;; value of pose (for retaining the loop value on nested
+                                    ;; calls). Revert it back after the execution of
+                                    ;; &body is completed.
+                                    (let ((old-current-pose current-pose))
+                                      (setf current-pose (funcall pseudo-pose value))
+                                      (unwind-protect
+                                           (progn ,@body)
+                                        (setf current-pose old-current-pose)))
+
+                                    ;; If no &body clause make the ik-service call
+                                    (call-ik-service
+                                     (funcall pseudo-pose value)
+                                     base-link-evaled
+                                     tip-link-evaled
+                                     seed-state-msg-evaled))
+
+                              ;; When a solution is obtained parse accordingly
+                              (when (and solution-msg
+                                         (or (not solution-valid-p-evaled)
+                                             (funcall solution-valid-p-evaled solution-msg)))
+                                (return (values solution-msg
```

Comment:
I just noticed that this line is not returning the value to the higher levels in the nest, resulting only the outermost nested value being returned. I am investigating into this.

Reply:
This is fixed in the new commit